### PR TITLE
Add documentation for 1.6

### PR DESCRIPTION
**Pull request by msau42**
**What type of PR is this?**
/kind documentation

**What this PR does / why we need it**:
prepare for 1.6 release

**Which issue(s) this PR fixes**:

Fixes #

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note
NONE
```

### CODE REVIEW

File: CHANGELOG-1.6.md [hunk 1]
```diff
@@ -0,0 +1,21 @@
+# Changes since v1.5.0
```
Comment (msau42) [inline CHANGELOG-1.6.md]: I should probably add deprecation notices for everything we plan to break with 2.0

**Comment by davidz627** [comment]
Please wait on this for a little bit. I have one more thing I need to add (the tiny companion fix for the migration bug fix)

**Comment by davidz627** [comment]
Depends on: https://github.com/kubernetes-csi/external-provisioner/pull/412

**Comment by jsafrane** [comment]
Can it wait for upcoming cloning fixes?

File: CHANGELOG-1.6.md [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+  ([#295](https://github.com/kubernetes-csi/external-provisioner/issues/295))
+- The "--leader-election-type" argument will be removed in a future release.
+  Only Leases-type leader election will be supported. The
+  "--enable-leader-election" argument will also be renamed to
+  "--leader-election".
+  ([#401](https://github.com/kubernetes-csi/external-provisioner/issues/401))
+- The default value of "--worker-threads" will be lowered to 10 to match the
+  other CSI sidecars in a future release.
+  ([#322](https://github.com/kubernetes-csi/external-provisioner/issues/322))
+- The external-provisioner will no longer default an empty fstype to "ext4".
```
[the remark this answers is not in the document]
Comment (msau42) [reply CHANGELOG-1.6.md]: stale comment whoops

File: CHANGELOG-1.6.md [hunk 3]
```diff
@@ -0,0 +1,64 @@
...
+  A new option to set the default will be added, otherwise the CSI driver
+  must be able to handle an empty fstype according to the CSI spec.
+  ([#328](https://github.com/kubernetes-csi/external-provisioner/issues/328))
```
Comment (xing-yang) [inline CHANGELOG-1.6.md]: Why are there referencing issues, not PRs?

Comment (msau42) [reply CHANGELOG-1.6.md]: The deprecations section is talking about behavior that we want to change in a future release, there are no PRs existing to implement them yet.

**Comment by msau42** [comment]
I've updated to sync up the latest changes and fix formatting issues. There's still 2 more prs we're waiting on, but there's a lot of content here so just wanted to kick off the review

**Comment by msau42** [comment]
This is ready to go
/hold cancel

**Pull request merged**